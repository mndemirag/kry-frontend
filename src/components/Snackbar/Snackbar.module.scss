@import "../../utils.scss";

.snackbar {
  visibility: hidden;
  width: 100%;
  max-width: 720px;
  background-color: $kry-blue;
  color: #fff;
  text-align: center;
  border-radius: 2px;
  padding: 16px;
  position: fixed;
  z-index: 1;
  bottom: 30px;
  font-size: 17px;
  position: fixed;
  left: 50%;
  transform: translateX(-50%);
}

.show {
  visibility: visible;
  -webkit-animation: fadein 0.5s, fadeout 0.5s 2.5s;
  animation: fadein 0.5s, fadeout 0.5s 2.5s;
}
