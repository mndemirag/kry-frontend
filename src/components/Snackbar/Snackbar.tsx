import { SnackbarProps } from "../../types";

import styles from "./Snackbar.module.scss";

export const Snackbar: React.FC<SnackbarProps> = ({ children, open }) => {
  return (
    <>
      <div className={`${styles.snackbar} ${open ? `${styles.show}` : ""}`}>
        {children}
      </div>
    </>
  );
};
