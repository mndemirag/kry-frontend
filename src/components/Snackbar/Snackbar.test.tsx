import { render } from "@testing-library/react";

import { Snackbar } from "./Snackbar";

const mockClose = jest.fn();

describe("Snackbar", () => {
  it("renders properly", () => {
    const { container } = render(<Snackbar onClose={mockClose} open={false} />);
    expect(container).toMatchSnapshot();
  });
});
