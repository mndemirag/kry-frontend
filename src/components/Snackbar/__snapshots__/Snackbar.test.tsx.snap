// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar renders properly 1`] = `
<div>
  <div
    class="snackbar "
  />
</div>
`;
